File: https:/github.com/rajesh-kayal-dev/Leetcode.git/121-BestTimeToBuyAndSellStock/121-BestTimeToBuyAndSellStock.java
// Last updated: 6/15/2026, 10:02:14 AM
class Solution {
    public int maxProfit(int[] prices) {

        int min_price=prices[0], profit=0;

        for(int i = 0; i <prices.length;i++){
            if(prices[i] < min_price){
                min_price= prices[i];
            }else{
                profit= Math.max(profit, prices[i]- min_price);
            }
        }
        return profit;
        
    }
}

File: https:/github.com/rajesh-kayal-dev/Leetcode.git/121-BestTimeToBuyAndSellStock/121-BestTimeToBuyAndSellStock.js
// Last updated: 6/15/2026, 10:02:20 AM
/**
 * @param {number[]} prices
 * @return {number}
 */
var maxProfit = function(prices) {
    let maxProfit = 0;
    let min = prices[0];

    for(let i = 0; i < prices.length; i++){
        if(prices[i] < min)min = prices[i];
            let profit = prices[i] - min;
         maxProfit = Math.max(profit, maxProfit);
        
    }
    return maxProfit;
};